nodejs/device/src/index.ts

```diff
@@ -248,123 +267,160 @@ class DeviceApp {
     );
   }
 
-  // Does not deal with out-of-order messages.
-  // For example, if we get index 5 and then 7, we assume 6 is lost.
-  private handleTestC2dMessage(obj: ThiefC2dMessage, msg: Message) {
-    if (
-      Number.isNaN(obj.thief.testC2dMessageIndex) ||
-      typeof obj.thief.testC2dMessageIndex !== "number"
-    ) {
-      this.client.reject(msg).catch(this.handleRejectC2dError);
-      return logger.warn("Issue with testC2dMessageIndex property. Rejecting.");
-    }
+  private handleC2dMessages(): [Promise<void>, () => void] {
```

Comment:
>handleC2dMessages [](start = 10, length = 17)

Here's where documentation might help.  I think this promise handles a single C2d message after it's been received.  The name implies that it handle more than one message.  I'm also a little unsure because the verb "handle" is very vague.  There's a school of thought that says "If you can't come up with a more specific verb, then that means that you can't describe what the code does, and maybe it needs to be refactored."  I don't think that's the case here, but "handle" as a verb is a smell that I watch out for.